src/main/java/org/alliancegenome/curation_api/jobs/executors/Gff3Executor.java

```diff
@@ -0,0 +1,190 @@
+import org.alliancegenome.curation_api.response.APIResponse;
+import org.alliancegenome.curation_api.response.LoadHistoryResponce;
+import org.alliancegenome.curation_api.services.CodingSequenceService;
+import org.alliancegenome.curation_api.services.ExonService;
+import org.alliancegenome.curation_api.services.Gff3Service;
+import org.alliancegenome.curation_api.services.TranscriptService;
+import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+import org.apache.commons.lang3.ObjectUtils;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class Gff3Executor extends LoadFileExecutor {
+
+	@Inject Gff3Service gff3Service;
+	@Inject ExonService exonService;
+	@Inject CodingSequenceService cdsService;
+	@Inject TranscriptService transcriptService;
+	
+	public void execLoad(BulkLoadFile bulkLoadFile) {
+		try {
+
+			CsvSchema gff3Schema = CsvSchemaBuilder.gff3Schema();
+			CsvMapper csvMapper = new CsvMapper();
+			MappingIterator<Gff3DTO> it = csvMapper.enable(CsvParser.Feature.INSERT_NULLS_FOR_MISSING_COLUMNS).readerFor(Gff3DTO.class).with(gff3Schema).readValues(new GZIPInputStream(new FileInputStream(bulkLoadFile.getLocalFilePath())));
+			List<Gff3DTO> gffData = it.readAll();
+			List<String> gffHeaderData = new ArrayList<>();
+			for (Gff3DTO gffLine : gffData) {
+				if (gffLine.getSeqId().startsWith("#")) {
+					gffHeaderData.add(gffLine.getSeqId());
+				} else {
+					break;
+				}
+			}
+			gffData.subList(0, gffHeaderData.size()).clear();
+
+			
+			BulkFMSLoad fmsLoad = (BulkFMSLoad) bulkLoadFile.getBulkLoad();
+			BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(fmsLoad.getFmsDataSubType());
+
+			Map<String, List<Long>> idsAdded = new HashMap<String, List<Long>>();
+			idsAdded.put("Transcript", new ArrayList<Long>());
+			idsAdded.put("Exon", new ArrayList<Long>());
+			idsAdded.put("CodingSequence", new ArrayList<Long>());
+			
+			Map<String, List<Long>> previousIds = getPreviouslyLoadedIds(dataProvider);
+			
+			BulkLoadFileHistory history = new BulkLoadFileHistory(gffData.size());
+			createHistory(history, bulkLoadFile);
+			idsAdded = runLoad(history, gffHeaderData, gffData, idsAdded, dataProvider);
+			runCleanup(transcriptService, history, dataProvider.name(), previousIds.get("Transcript"), idsAdded.get("Transcript"), "GFF transcript", bulkLoadFile.getMd5Sum());
+			runCleanup(exonService, history, dataProvider.name(), previousIds.get("Exon"), idsAdded.get("Exon"), "GFF exon", bulkLoadFile.getMd5Sum());
+			runCleanup(cdsService, history, dataProvider.name(), previousIds.get("CodingSequence"), idsAdded.get("CodingSequence"), "GFF coding sequence", bulkLoadFile.getMd5Sum());
+			
```

Comment:
I wonder if it makes sense to add a `runCleanup` overload that can handle maps of lists so that `idsAdded` and `previousIds` maps could just be passed into one call?

Reply:
They all need different services though

Reply:
Ah, yeah. That's a good point.

Reply:
We could write a method to do all that and pass the appropriate stuff to the generic runCleanup, but I think it might be more confusing than just having multiple calls.

Reply:
Yeah, that's probably right. We could have like a map of maps that bundles the lists together with the appropriate service, but that's probably just overengineering at that point.